src/csharp/examples/server/API/DoCoolStuff/DoCoolStuffController.cs

```diff
@@ -0,0 +1,61 @@
+using Microsoft.AspNetCore.Mvc;
+using System;
+using SessionlessExample.Server.Models;
+using System.Text.Json;
+using SessionlessNET.Impl;
+using SessionlessNET.Models;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace SessionlessExample.Server.Controllers
+
+{
+    [ApiController]
+    [Route("/cool-stuff")]
+    public class DoCoolStuffController: ControllerBase
+    {
+        Vault Vault;
+        Sessionless Sessionless;
+        const string privPath = "./priv.key";
+        const string pubPath = "./pub.key";
+        static void VaultSaver(KeyPairHex pair) {
+            
+        }
+        static KeyPairHex? VaultGetter() {
+            
+            return new("", "");
+        }
+
+        [HttpPost]
+        [Consumes("application/json")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status403Forbidden)]
+        public IActionResult PostJson(Models.DoCoolStuffModel input) {
+            Vault = new Vault(VaultGetter, VaultSaver);
+            Sessionless = new(Vault);
+
+            UserModel user = DemoPersistenceController.getUser(input.uuid);
+
+            if(user == null) {
+                return Problem("No user found");
+            }
+                        
+            var postedSignature = Request.Headers["Signature"];
+
+            MessageSignatureHex signature = new MessageSignatureHex(postedSignature);
```

Comment:
`postedSignature` could be null

and a suggestion:
C# encourages to use `new()` rather than `new Thin()` when the type is obvious 
```csharp
 MessageSignatureHex signature = new(postedSignature);
 //or implicit inference 
 var signature = new MessageSignatureHex(postedSignature);
 ```